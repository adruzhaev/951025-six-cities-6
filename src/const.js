export const CITIES = [
  {
    id: 1,
    city: `Paris`
  },
  {
    id: 2,
    city: `Cologne`
  },
  {
    id: 3,
    city: `Brussels`
  },
  {
    id: 4,
    city: `Amsterdam`
  },
  {
    id: 5,
    city: `Hamburg`
  },
  {
    id: 6,
    city: `Dusseldorf`
  }
];

export const RATING = [
  {
    value: 5,
    title: `perfect`,
  },
  {
    value: 4,
    title: `good`,
  },
  {
    value: 3,
    title: `not bad`,
  },
  {
    value: 2,
    title: `badly`,
  },
  {
    value: 1,
    title: `terribly`,
  },
];

export const SortingTypes = {
  POPULAR: `Popular`,
  PRICE_DESCENDING: `Price: low to high`,
  PRICE_ASCENDING: `Price: high to low`,
  TOP_FIRST: `Top rated first`,
};
