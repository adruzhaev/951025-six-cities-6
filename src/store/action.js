export const ActionType = {
  CHANGE_CITY: `city/change`,
  CHANGE_SORTING: `sorting/change`,
  LOAD_OFFERS: `data/loadOffers`,
  LOAD_OFFER: `data/loadOffer`,
  REQUIRED_AUTHORIZATION: `user/requiredAuthorization`,
  SET_AUTHORIZATION: `user/authInfo`,
};

export const ActionCreator = {
  changeCity: (city) => ({
    type: ActionType.CHANGE_CITY,
    payload: city,
  }),
  changeSorting: (sortingType) => ({
    type: ActionType.CHANGE_SORTING,
    payload: sortingType,
  }),
  loadOffers: (offers) => ({
    type: ActionType.LOAD_OFFERS,
    payload: offers,
  }),
  loadOffer: (offer) => ({
    type: ActionType.LOAD_OFFER,
    payload: offer,
  }),
  requireAuthorization: (status) => ({
    type: ActionType.REQUIRED_AUTHORIZATION,
    payload: status,
  }),
  setAuthorization: (info) => ({
    type: ActionType.SET_AUTHORIZATION,
    payload: info,
  }),
};
