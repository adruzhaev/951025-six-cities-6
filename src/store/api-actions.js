import {AuthorizationStatus, APIRoutes, AppRoutes} from '../const';
import {ActionCreator} from './action';
// import browserHistory from '../browser-history';

const adaptToClient = (item) => {
  const adaptedOffer = Object.assign(
      {},
      item,
      {
        isPremium: item.is_premium,
        isFavorite: item.is_favorite,
        previewImage: item.preview_image,
        maxAdults: item.max_adults,
        host: {
          ...item.host,
          avatarUrl: item.host.avatar_url
        }
      }
  );

  delete adaptedOffer.preview_image;
  delete adaptedOffer.is_favorite;
  delete adaptedOffer.is_premium;
  delete adaptedOffer.max_adults;
  delete adaptedOffer.host.avatar_url;

  return adaptedOffer;
};

export const fetchOffersList = () => (dispatch, _getState, api) => (
  api.get(APIRoutes.OFFERS)
    .then(({data}) => dispatch(ActionCreator.loadOffers(data.map((item) => adaptToClient(item)))))
);

export const fetchOffer = (id) => (dispatch, _getState, api) => (
  api.get(APIRoutes.OFFERS + `/` + id)
    .then(({data}) => dispatch(ActionCreator.loadOffer(adaptToClient(data))))
);

export const checkAuth = () => (dispatch, _getState, api) => (
  api.get(APIRoutes.LOGIN)
    .then(({data}) => dispatch(ActionCreator.setAuthorization(data)))
    .then(() => dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH)))
    .catch(() => {})
);

export const login = ({email, password}) => (dispatch, _getState, api) => (
  api.post(APIRoutes.LOGIN, {email, password})
    .then(({data}) => dispatch(ActionCreator.setAuthorization(data)))
    .then(() => dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH)))
    .then(() => dispatch(ActionCreator.redirectToRoute(AppRoutes.MAIN)))
    .catch((error) => error)
);
