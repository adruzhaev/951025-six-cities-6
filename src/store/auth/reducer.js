import {ActionType} from './action';
import {AuthorizationStatus} from '../../const';

const initialState = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  authorizationInfo: {},
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.REQUIRED_AUTHORIZATION:
      return {
        ...state,
        authorizationStatus: action.payload
      };
    case ActionType.SET_AUTHORIZATION:
      return {
        ...state,
        authorizationInfo: action.payload
      };
    default:
      return state;
  }
};

export {reducer};
