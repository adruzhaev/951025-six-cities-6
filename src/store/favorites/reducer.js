import {ActionType} from './action';

const initialState = {
  offers: [],
  isDataLoaded: false,
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.ADD_FAVORITE:
      return {
        ...state,
        offers: state.isDataLoaded ? [...state.offers, action.payload] : []
      };
    case ActionType.REMOVE_FAVORITE:
      return {
        ...state,
        offers: state.offers.filter((offer) => offer.id !== action.payload.id)
      };
    case ActionType.LOAD_FAVORITES:
      return {
        ...state,
        offers: action.payload,
        isDataLoaded: true,
      };
    default:
      return state;
  }
};

export {reducer};
