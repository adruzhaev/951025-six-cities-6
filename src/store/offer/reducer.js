import {ActionType} from './action';
import {CITIES, SortingTypes, OfferStatus} from '../../const';

const initialState = {
  activeCity: CITIES[0].city,
  activeSorting: SortingTypes.POPULAR,
  offers: [],
  offer: null,
  isDataLoaded: false,
  notFoundOffer: OfferStatus.FOUND,
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.CHANGE_CITY:
      return {
        ...state,
        activeCity: action.payload
      };
    case ActionType.CHANGE_SORTING:
      return {
        ...state,
        activeSorting: action.payload
      };
    case ActionType.LOAD_OFFERS:
      return {
        ...state,
        offers: action.payload,
        isDataLoaded: true,
      };
    case ActionType.LOAD_OFFER:
      return {
        ...state,
        offer: action.payload,
      };

    case ActionType.CLEAN_STATE:
      return {
        ...state,
        notFoundOffer: OfferStatus.FOUND,
        offer: null,
      };

    case ActionType.NOT_FOUND_OFFER:
      return {
        ...state,
        notFoundOffer: OfferStatus.NOT_FOUND,
      };
    default:
      return state;
  }
};

export {reducer};
