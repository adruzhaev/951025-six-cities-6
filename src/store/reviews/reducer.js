import {ActionType} from './action';

const initialState = {
  reviews: [],
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.LOAD_REVIEWS:
      return {
        ...state,
        reviews: action.payload,
      };
    default:
      return state;
  }
};

export {reducer};
