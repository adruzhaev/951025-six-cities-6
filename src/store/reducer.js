import {combineReducers} from 'redux';
import {reducer as auth} from './auth/reducer';
import {reducer as offer} from './offer/reducer';
import {reducer as reviews} from './reviews/reducer';

const reducer = combineReducers({
  auth,
  offer,
  reviews
});

export {reducer};
