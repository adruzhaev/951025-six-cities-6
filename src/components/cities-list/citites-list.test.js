import React from 'react';
import {render} from '@testing-library/react';
import {Router} from 'react-router-dom';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {createMemoryHistory} from 'history';
import {NameSpace} from '../../store/reducer';
import CitiesList from './cities-list.jsx';
import {CITIES} from '../../const';

const mockStore = configureStore();

it(`CitiesList should render correctly`, () => {
  const history = createMemoryHistory();
  const {container} = render(
      <Provider store={mockStore({
        [NameSpace.OFFER]: {
          activeCity: CITIES[0].city,
        }
      })}>
        <Router history={history}>
          <CitiesList />
        </Router>
      </Provider>
  );
  expect(container).toMatchSnapshot();
});
