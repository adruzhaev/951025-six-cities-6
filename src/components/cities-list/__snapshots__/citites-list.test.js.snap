// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitiesList should render correctly 1`] = `
<div>
  <ul
    class="locations__list tabs__list"
  >
    <li
      class="locations__item"
    >
      <span
        class="locations__item-link tabs__item tabs__item--active"
      >
        Paris
      </span>
    </li>
    <li
      class="locations__item"
    >
      <span
        class="locations__item-link tabs__item "
      >
        Cologne
      </span>
    </li>
    <li
      class="locations__item"
    >
      <span
        class="locations__item-link tabs__item "
      >
        Brussels
      </span>
    </li>
    <li
      class="locations__item"
    >
      <span
        class="locations__item-link tabs__item "
      >
        Amsterdam
      </span>
    </li>
    <li
      class="locations__item"
    >
      <span
        class="locations__item-link tabs__item "
      >
        Hamburg
      </span>
    </li>
    <li
      class="locations__item"
    >
      <span
        class="locations__item-link tabs__item "
      >
        Dusseldorf
      </span>
    </li>
  </ul>
</div>
`;
